### PR TITLE
GitHub comment bot commands

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Add GH comment to re-trigger the build. 

## Why is it important?

Engage with the PR contributors to facilitate them how to interact with the CI

## Important

It only supports the pattern `^/test` for the time being, to avoid re-triggering builds over and over if a PR comment is added with the comment.

## UI

image

image

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-@Library('apm@master') _
+@Library('apm@test/github-comment-bot') _
```
Comment (v1v) [inline .ci/Jenkinsfile:18]: For testing purposes only

**Review by sqren — APPROVED** [review]

File: .ci/Jenkinsfile (hunk 1 above)
Comment (v1v) [reply .ci/Jenkinsfile:18]:
```suggestion
@Library('apm@master') _
```

File: src/co/elastic/NotificationManager.groovy [hunk 2]
```diff
@@ -414,3 +417,35 @@ def queryFilter(jobName) {
...
+
+/**
+ * This method searches for the IssueCommentTrigger in the project itself
+ * and if so, then look for the GitHub comment trigger which it's supported.
+ */
+def issueCommentTrigger() {
+  def issueCommentTrigger = findIssueCommentTrigger()
+
+  if (issueCommentTrigger == null) {
+    log(level: 'WARN', text: "No comment bad happened when creating")
```
Comment (mdelapenya) [inline src/co/elastic/NotificationManager.groovy:430]: `bad happened`? Not sure if `has / had` happened :)

Comment (v1v) [reply src/co/elastic/NotificationManager.groovy:430]:
```suggestion
    log(level: 'WARN', text: "No comment has happened when creating")
```

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**